### ISSUE TITLE
[Bug]: Disabling ipv4_enabled on a server restarts the server without warning

### ISSUE THREAD

**Issue by nomorepanic** [hetznercloud/terraform-provider-hcloud#999]
### What happened?

When ipv4_enabled in a server, the server gets restarted without warning.

It's customary for providers to require an additional field / confirmation for these kind of operations, for example by having a force_restart option in the public_net section.

Ran with hcloud v1.48.1 on tofu 1.8.1

### What did you expect to happen?

I expected tofu to refuse the change!

### Please provide a minimal working example

All you have to do is:

```hcl
resource "hcloud_server" "myserver" {
  ...
  public_net {
    ipv4_enabled = true
    ipv6_enabled = true
  }
}
```

Plan and apply, and then:

```hcl
resource "hcloud_server" "myserver" {
  ...
  public_net {
    ipv4_enabled = false
    ipv6_enabled = true
  }
}
```

To see you server get restarted without warning.

**Comment by jooola** [comment]
The backend does not support attaching or detaching Primary IPs without stopping the server. Therefor, we automatically shut down the server to be able to apply those changes.

Adding a warning is a good idea, but it will require some work to make sure we only shut down / power on the server only once, which is currently not the case.

**Comment by jooola** [comment]
Closing this as duplicate of #908
